### DISCUSSION TITLE
Using Expo API Routes feature with tRPC?

### DISCUSSION THREAD

**Discussion by ax-at** [expo/expo#26857] [Archived: Questions]
Is there a possibility to merge "tRPC Express Adapter" with "Expo API Route Express Deployment" in a single project?

tRPC Express Adapter: https://trpc.io/docs/server/adapters/express
Expo API Route Express Deployment: https://docs.expo.dev/router/reference/api-routes/#express

**Comment by ax-at** [comment]
I think, I was in a hurry when this thought came to my mind and I posted this idea here, without understanding the nature of both "Expo API Routes" and "tRPC". I think both are serving different kind of approaches and with current implementations, I think it won't be a good idea to integrate them both right now!

Thanks.

**Comment by Poggen** [comment]
I thought the same thing and I am working on this now. I have trpc working based on the fabulous work from @gsporto in https://github.com/gsporto/expo-api-routes-trpc, but I am struggling to get express working with expo. My use-case is that I, for various reasons, would like to deploy the server functions to firebase, which leaves express as the only option.

**Reply by mgordillo11** [reply]
Were you ever able to figure out how to use express with Expo? I'm running in a similar situation, thanks in advance!

**Reply by gsporto** [reply]
Hi thanks for the mention @Poggen, did you guys tried this https://docs.expo.dev/router/reference/api-routes/#express ? If yes and didn't work, give me feedback, so I can look into it.
